deploy/deploy-api.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+#
+# Deploy API Server on GCP.
+
+if (( $# != 1 ))
+then
+  echo "Usage: deploy/deploy-api.sh <dataset>"
+  echo "  where <dataset> is the name of a directory in api/dataset_config/"
+  echo "Run this script from project root"
+  exit 1
+fi
+
+dataset=$1
+project_id=$(grep -v "//" api/dataset_config/${dataset}/deploy.json | jq --raw-output '.project_id')
```

Comment:
"^//"?

Reply:
I get:
```
parse error: Invalid numeric literal at line 2, column 5
```

Reply:
Maybe escape the forward slashes? The gist of the comment was to anchor the regex to only `grep -v` lines that start with `//`

Reply:
Actually, I realized this doesn't make sense. Say the json file has:
```
  "project_id": "test-data-explorer"   // some comment
```
You're suggesting I leave that line in. But then jq will fail because the comment isn't valid json.

The right approach would be to strip only the comment. I don't think it's worth the effort to figure that out. (Note: jsmin handles this case correctly, but I don't want to make the user install jsmin.)

Reply:
This overhead is why I initially pushed to not have comments in our json. However, if we allow comments here, we should handle this case, because if we choose to leave the line in or remove it totally, our program will fail (in a very opaque way, as a user I'd have a very tough time tracking down `grep -v "//"` as the root cause of my `missing project_id` error). However, I also agree it's not worth it to write our own json parser. What do you think about changing our json config to yaml instead? That supports comments and is generally easier on the eyes when it comes to human readability.

Reply:
I prefer json because it's widely used within the community. How about I add a comment to deploy.json saying that comments must begin at the beginning of a line?

Reply:
I think regardless of where we settle we should be consistent throughout the way we treat comments in json. So if we remove only whole line comments here, we should do the same in our indexers, etc.

Is it really worth having comments in our json to the point where we need to write comments about our comment structure? This seems like it's adding a fair amount of complexity on this:
```
{
  "project_id": "GCP_PROJECT_ID"
}
```
To me, this seems sufficiently self explanatory without the need for comments. I'm not sure exactly what you had in mind for deploy.json, but if it's just the project_id there's other ways to go about it and I'm not sure we need a separate config file for that.d

Reply:
I'll remove the comment from deploy.json in next PR. If I want to add a comment later, let's discuss in person.

Reply:
`-v "//"` no longer required here, at least for the time being

Reply:
Done. Oops, forgot to push some previous fixes, pushed now